html/webappapis/system-state-and-capabilities/the-navigator-object/NavigatorID.js

```diff
@@ -60,4 +53,51 @@ function run_test() {
                         "filter_name=User-Agent");
     request.send();
   }, "userAgent value");
+
+  test(function() {
+    if ("window" in self) {
+      if (navigator.userAgent.indexOf("Chrome") != -1) {
+        // "If the navigator compatibility mode is Chrome"
+        assert_equals(navigator.vendor, "Google Inc.");
+      } else if (navigator.userAgent.indexOf("WebKit") != -1) {
+        // "If the navigator compatibility mode is WebKit"
+        assert_equals(navigator.vendor, "Apple Computer, Inc.");
+      } else {
+        // "If the navigator compatibility mode is Gecko"
+        assert_equals(navigator.vendor, "");
+      }
+    } else {
+      assert_false("vendor" in navigator);
+    }
+  }, "vendor");
+
+  test(function() {
+    if ("window" in self) {
+      assert_equals(navigator.vendorSub, "");
+    } else {
+      assert_false("vendorSub" in navigator);
+    }
+  }, "vendorSub");
+
+  // "If the navigator compatibility mode is Gecko, then the user agent must
+  // also support the following partial interface" (taintEnabled() and oscpu)
+  // See https://www.w3.org/Bugs/Public/show_bug.cgi?id=22555 and
+  // https://www.w3.org/Bugs/Public/show_bug.cgi?id=27820
+
+  test(function() {
+    if ("window" in self && navigator.userAgent.indexOf("WebKit") == -1) {
```

Comment:
This file uses several different ways of testing navigator compatibility mode, not all of which are compatible. E.g. here Gecko means "does not have the string WebKit", where above Gecko means "does not have the string WebKit or the string Chrome".

It would be good to factor this out into a variable (in the test file) that is one of "Chrome", "WebKit", or "Gecko", and then the test code proper could say `if ("window" in self && navigatorCompatibilityMode === "Gecko")`. Then you wouldn't need the spec quotes as much.